src/rgw/rgw_putobj_processor.h

```diff
@@ -25,43 +25,30 @@
 
 namespace rgw {
 
+namespace sal {
+  class RadosStore;
+}
+
 class Aio;
 
 namespace putobj {
 
-// a data consumer that writes an object in a bucket
-class ObjectProcessor : public DataProcessor {
- public:
-  // prepare to start processing object data
-  virtual int prepare(optional_yield y) = 0;
-
-  // complete the operation and make its result visible to clients
-  virtual int complete(size_t accounted_size, const std::string& etag,
-                       ceph::real_time *mtime, ceph::real_time set_mtime,
-                       std::map<std::string, bufferlist>& attrs,
-                       ceph::real_time delete_at,
-                       const char *if_match, const char *if_nomatch,
-                       const std::string *user_data,
-                       rgw_zone_set *zones_trace, bool *canceled,
-                       optional_yield y) = 0;
-};
-
 // an object processor with special handling for the first chunk of the head.
 // the virtual process_first_chunk() function returns a processor to handle the
 // rest of the object
-class HeadObjectProcessor : public ObjectProcessor {
+class HeadObjectProcessor : public rgw::sal::ObjectProcessor {
```

Comment:
nit: this is under `namespace rgw` already, so this can just be `sal::ObjectProcessor`

Reply:
I never liked partial namespacing.  It confuses tagging.